utils/errors/errors.go

```diff
@@ -0,0 +1,127 @@
+// Package errors contains a custom HTTP error handler that adds an extra `Error` field to error payloads.
+// This helps us maintain compatibility with older PMM clients while keeping the benefits of grpc-gateway v2.
+// Our requirement is fairly minimal, but copy a lot of code from grpc-gateway to account for unexported fields and methods
+package errors
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/textproto"
+	"strings"
+
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/grpclog"
+	"google.golang.org/grpc/status"
+
+	"github.com/percona/pmm/api/serverpb"
+)
+
+// PMMHTTPErrorHandler is a custom implementation of DefaultHTTPErrorHandler with an `Extra` error field to support old PMM clients
+func PMMHTTPErrorHandler(ctx context.Context, mux *runtime.ServeMux, marshaler runtime.Marshaler, w http.ResponseWriter, r *http.Request, err error) {
```

Comment:
Please add link to https://github.com/grpc-ecosystem/grpc-gateway/blob/aec6aa29864109e41408491319a859f190ec4040/runtime/errors.go#L93
to comment.